packages/vxrn/src/utils/patches.ts

```diff
@@ -127,44 +160,50 @@ export async function applyDependencyPatches(
               }
 
               const filesToApply = file.includes('*') ? globDir(nodeModuleDir, file) : [file]
-              const appliedContents = new Map<string, string>()
 
               await Promise.all(
                 filesToApply.map(async (relativePath) => {
                   try {
                     const fullPath = join(nodeModuleDir, relativePath)
-                    const ogFile = fullPath + '.vxrn.ogfile'
-
-                    // for any update we store an "og" file to compare and decide if we need to run again
-                    let existingPatch: string | null = appliedContents.get(ogFile) || null
-
-                    if (!existingPatch) {
-                      if (!process.env.VXRN_FORCE_PATCH) {
-                        if (FSExtra.existsSync(ogFile)) {
-                          try {
-                            // for some reason with bun install this would say it exists? but then fail here?
-                            existingPatch = await FSExtra.readFile(ogFile, 'utf-8')
-                          } catch (err) {
-                            console.warn(`Error reading patch`, err)
-                          }
-                        }
-                      }
+
+                    if (!process.env.VXRN_FORCE_PATCH && isAlreadyPatchedMap.get(fullPath)) {
+                      // if the file is already patched, skip it
+                      return
                     }
 
-                    let contentsIn =
-                      existingPatch ||
-                      (FSExtra.existsSync(fullPath)
-                        ? await FSExtra.readFile(fullPath, 'utf-8')
-                        : '')
+                    let contentsIn = await (async () => {
+                      if (pathsBeingPatched.has(fullPath)) {
+                        // If the file has been patched during the current run,
+                        // we should always start from the already patched file
+                        return await FSExtra.readFile(fullPath, 'utf-8')
```

Comment:
idk if this may regress my fix because we want to start from the previously patched contents not the original one. My issue you can see replicated in chat app as we have a custom reanimated patch that must run after the patch we apply internally. maybe it works but not sure

Reply:
I think it should still work, because after each patch, the patched content should be written to the disk. On the next patch, what we read here will be the output of the previous patch.